donate/views.py

```diff
@@ -399,3 +431,58 @@ def school_donation(request):
     else:
         form = SchoolDonationForm()
     return render(request, 'donate/school_donation.html', {'form': form})
+
+
+@csrf_exempt
+def ilw_payment_callback(request):
+    context = {}
+    status_template = 'spoken/templates/ilw_payment_status.html'
+    order_id = request.POST.get('order_id')
+    if not order_id:
+        raise Http404
+    
+    payee = Payee.objects.get(transaction__order_id=order_id)
+    context['order_id'] = order_id
+    try:
+        order_status_url = f"{settings.ORDER_STATUS_URL}{order_id}"
+        # order status api
+        headers = get_request_headers(payee.email)
+        try:
+            response = requests.get(order_status_url, headers=headers)
+            response_data = response.json()
+        except requests.exceptions.RequestException as e:
+            context['status'] = 'FAILED'
+            return render(request, status_template, context=context)
+        if response.status_code == 200:
+            verified = verify_hmac_signature(request.POST)
+            if not verified:
+                context['status'] = 'FAILED'
+                return render(request, status_template, context=context)
+            save_ilw_hdfc_success_data(order_id, response_data)
+            data = get_display_transaction_details(response_data)
+            data['udf5'] = response_data.get('udf5', '')
+            context['data'] = data
+            order_status = response_data.get('status', '')
+            amount = response_data.get('amount', '')
+            try:
+                amount_decimal = Decimal(str(amount))
+            except InvalidOperation:
+                context['status'] = 'FAILED'
+                return render(request, status_template, context=context)
```

Comment:
The try should be inside the if order_status == 'CHARGED' case. when status is say PENDING, the amount may not be available.